presentation/src/main/java/com/sopt/presentation/appointmentCreate/appointmentCreateTimePicker/AppointmentCreateTimePickerRoute.kt

```diff
@@ -0,0 +1,276 @@
+                    style = typography.b4SemiBold,
+                    color = colors.gray900
+                )
+                CircularCheckbox(
+                    isChecked = isChecked,
+                    onCheckedChange = {
+                        isChecked = it
+                        showPicker = !it
+                        if (isChecked) {
+                            selectedStartHour = null
+                            selectedEndHour = null
+                        }
+                    }
+                )
+            }
+
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 32.dp, bottom = 9.dp),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                NoostakSubHeaderText(
+                    stringResource(R.string.text_calendar_appointment_time_check),
+                    modifier = Modifier.weight(1f)
+                )
+                Box(
+                    modifier = Modifier.size(24.dp),
+                    contentAlignment = Alignment.Center
+                ) {
+                    Image(
+                        imageVector = ImageVector.vectorResource(
+                            id = if (isChecked) R.drawable.ic_calendar_off else R.drawable.ic_calendar_on
+                        ),
+                        contentDescription = null,
+                        modifier = Modifier.fillMaxSize()
+                    )
+                }
+            }
+
+            if (showPicker) {
+                NoostakTimePicker { startHour, endHour ->
+                    selectedStartHour = startHour
+                    selectedEndHour = endHour
+                }
+            }
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            NoostakBottomButton(
+                text = stringResource(R.string.text_calendar_appointment_next),
+                onButtonClick = {
+                    val adjustedEndHour = if (selectedEndHour == 0) 24 else selectedEndHour ?: 24
+                    val duration = selectedStartHour?.let { startHour ->
+                        if (startHour < adjustedEndHour) {
+                            adjustedEndHour - startHour
+                        } else {
+                            adjustedEndHour + 24 - startHour
+                        }
+                    } ?: 0
+
+                    if (selectedStartHour == selectedEndHour) {
+                        scope.launch {
+                            snackbarHostState.showSnackbar("24시간 이하로 선택해주세요")
+                        }
+                        return@NoostakBottomButton
+                    }
+
+                    if (duration < appointmentDuration) {
+                        scope.launch {
+                            snackbarHostState.showSnackbar("${appointmentDuration}시간 이상을 선택해주세요.")
+                        }
+                        return@NoostakBottomButton
+                    }
```

Comment:
P1: 현재 CircularCheckbox 클릭 시 selectedStartHour와 selectedEndHour가 null로 설정되므로, 
기존 조건문에 의해 스낵바가 나옵니다. 다음과 같이 null 체크해주세요

```suggestion
                    if (selectedStartHour != null && selectedEndHour != null) {
                        when {
                            selectedStartHour == selectedEndHour -> {
                                scope.launch {
                                    snackbarHostState.showSnackbar("24시간 이하로 선택해주세요")
                                }
                                return@NoostakBottomButton
                            }
                            duration < appointmentDuration -> {
                                scope.launch {
                                    snackbarHostState.showSnackbar("${appointmentDuration}시간 이상을 선택해주세요.")
                                }
                                return@NoostakBottomButton
                            }
                        }
                    }
```